src/lib/common/differenceSignal.cpp

```diff
@@ -16,6 +25,12 @@ bool DifferenceSignal::addSample(double sample) {
   return false;
 }
 
+/*!
+ * @brief Encodes the difference signal to a buffer
+ * @param d_n[out] The buffer to store the difference signal
+ * @param numSamples[in, out] In: The buffer sample size, Out: The number of samples encoded into the buffer
+ * @return True if the difference signal was encoded, false otherwise
+ */
 bool DifferenceSignal::encodeDifferenceSignalToBuffer(double *d_n, size_t &numSamples) {
```

Comment:
@russelldeguzman it looks like this DifferenceSignal class is responsible for allocating the buffer for storing the raw data, but the caller is responsible for allocating the buffer to store the output difference signal and passing it in here, yea?

Reply:
Yes that's correct!

Reply:
It also looks like it's OK to pass in any sized buffer we want - so if we only want 300 samples for our output we can pass in 300 even if there are 1500 samples in the raw buffer. We can pass in 2000 samples, and it will make sure to not run off the end of the raw buffer.
One thing that the caller is responsible for is making sure they have actually allocated enough memory in `d` per how many samples we request.